packages/tailwindcss/src/compat/screens-config.test.ts

```diff
@@ -0,0 +1,592 @@
+      }
+    }
+    .lg\\:flex {
+      @media (width >= 60rem) {
+        display: flex;
+      }
+    }
+    .print\\:items-end {
+      @media print {
+        align-items: flex-end;
+      }
+    }
+    "
+  `)
+})
+
+test('JS config `screens` only setup, even if those match the default-theme export', async () => {
+  let input = css`
+    @config "./config.js";
+    @tailwind utilities;
+  `
+
+  let compiler = await compile(input, {
+    loadConfig: async () => ({
+      theme: {
+        screens: {
+          sm: '40rem',
+          md: '48rem',
+          lg: '64rem',
+        },
+      },
+    }),
+  })
+
+  expect(
+    compiler.build([
+      // Order is messed up on purpose
+      'md:flex',
+      'sm:flex',
+      'lg:flex',
+      'min-md:max-lg:underline',
+      'min-sm:max-md:underline',
+
+      // Ensure other core variants appear at the end
+      'print:items-end',
+    ]),
+  ).toMatchInlineSnapshot(`
+    ".sm\\:flex {
+      @media (width >= 40rem) {
+        display: flex;
+      }
+    }
+    .min-sm\\:max-md\\:underline {
+      @media (width >= 40rem) {
+        @media (width < 48rem) {
+          text-decoration-line: underline;
+        }
+      }
+    }
+    .md\\:flex {
+      @media (width >= 48rem) {
+        display: flex;
+      }
+    }
+    .min-md\\:max-lg\\:underline {
+      @media (width >= 48rem) {
+        @media (width < 64rem) {
+          text-decoration-line: underline;
+        }
+      }
+    }
+    .lg\\:flex {
+      @media (width >= 64rem) {
+        display: flex;
+      }
+    }
+    .print\\:items-end {
+      @media print {
+        align-items: flex-end;
+      }
+    }
+    "
+  `)
+})
+
+test('JS config `screens` overwrite CSS `--breakpoint-*`', async () => {
+  let input = css`
+    @theme default {
+      --breakpoint-sm: 40rem;
+      --breakpoint-md: 48rem;
+      --breakpoint-lg: 64rem;
+      --breakpoint-xl: 80rem;
+      --breakpoint-2xl: 96rem;
+    }
+    @config "./config.js";
+    @tailwind utilities;
+  `
+
+  let compiler = await compile(input, {
+    loadConfig: async () => ({
+      theme: {
+        screens: {
+          mini: '40rem',
+          midi: '48rem',
+          maxi: '64rem',
+        },
+      },
+    }),
+  })
+
+  expect(
+    compiler.build([
+      'sm:flex',
+      'md:flex',
+      'mini:flex',
+      'midi:flex',
+      'maxi:flex',
+      'min-md:max-lg:underline',
+      'min-sm:max-md:underline',
+      'min-midi:max-maxi:underline',
+      'min-mini:max-midi:underline',
+
+      // Ensure other core variants appear at the end
+      'print:items-end',
+    ]),
+  ).toMatchInlineSnapshot(`
+    ":root {
+      --breakpoint-sm: 40rem;
+      --breakpoint-md: 48rem;
+      --breakpoint-lg: 64rem;
+      --breakpoint-xl: 80rem;
+      --breakpoint-2xl: 96rem;
+    }
+    .mini\\:flex {
+      @media (width >= 40rem) {
+        display: flex;
+      }
+    }
+    .sm\\:flex {
+      @media (width >= 40rem) {
+        display: flex;
+      }
+    }
```

Comment:
I think this is wrong. The `sm` variant shouldn't exist right — because the theme wasn't extended? (and as such the user is explicitly saying they don't want "default" screens values)

Reply:
Yeah that something me and @adamwathan chatted about and we decided that we want this behavior instead since it’s also how `—color` extension work right now. We append the user resolved theme values without throwing anyway anything that is defined in the CSS.

Reply:
👍